python/pyspark/sql/dataframe.py

```diff
@@ -1272,16 +1272,18 @@ def replace(self, to_replace, value, subset=None):
         """Returns a new :class:`DataFrame` replacing a value with another value.
         :func:`DataFrame.replace` and :func:`DataFrameNaFunctions.replace` are
         aliases of each other.
+        Values `to_replace` and `value` should be homogeneous. Mixed string and numeric
```

Comment:
It's true that it will succeed when passed a float for an int, but it chops the float into an int which I suspect would be confusing if anyone passed in something with a non-zero decimal component. If you look at the documentation in Scala side it says "Key and value of `replacement` map must have the same type"